great_expectations/datasource/batch_kwargs_generator/batch_kwargs_generator.py

```diff
@@ -225,11 +225,6 @@ def get_iterator(self, data_asset_name=None, **kwargs):
             return self._data_asset_iterators[data_asset_name][0]
 
     def build_batch_kwargs(self, data_asset_name=None, partition_id=None, **kwargs):
-        if (not kwargs.get("name") and not data_asset_name) or (
-            kwargs.get("name") and data_asset_name
-        ):
-            raise ValueError("Please provide either name or data_asset_name.")
-
```

Comment:
No explicit deprecation warning tag but this gets called downstream - we shouldn't support `name` here either